variables.virtualnetwork.tf

```diff
@@ -1,142 +1,113 @@
 
-If enabled the `vwan_hub_resource_id` must also be supplied.
-DESCRIPTION
-  default     = false
-}
+> Note at least one of `location` or `var.location` must be specified.
+> If both are empty then the module will fail.
 
-variable "vwan_hub_resource_id" {
-  type        = string
-  description = <<DESCRIPTION
-The resource ID of the vwan hub to which the virtual network will be connected.
-E.g. `/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/my-rg/providers/Microsoft.Network/virtualHubs/my-hub`
+### Hub network peering values
 
-Leave blank to create a virtual network without a vwan hub connection.
-DESCRIPTION
-  default     = ""
-}
+The following values configure bi-directional hub & spoke peering for the given virtual network.
 
-variable "virtual_network_resource_group_name" {
-  type        = string
-  description = <<DESCRIPTION
-The name of the resource group to create the virtual network in.
-DESCRIPTION
-  default     = ""
-}
+- `hub_peering_enabled`: Whether to enable hub peering. [optional]
+- `hub_network_resource_id`: The resource ID of the hub network to peer with. [optional - but required if hub_peering_enabled is `true`]
+- `hub_peering_name_tohub`: The name of the peering to the hub network. [optional - leave empty to use calculated name]
+- `hub_peering_name_fromhub`: The name of the peering from the hub network. [optional - leave empty to use calculated name]
+- `hub_peering_use_remote_gateways`: Whether to use remote gateways for the hub peering. [optional - default true]
 
-variable "virtual_network_location" {
-  type        = string
-  description = <<DESCRIPTION
-The location of the virtual network.
+### Mesh peering values
 
-Use this to override the default location defined by `var.location`.
-Leave blank to use the default location.
-DESCRIPTION
-  default     = ""
-}
+Mesh peering is the capability to create a bi-directional peerings between all supplied virtual networks in `var.virtual_networks`.
+Peerings will only be created between virtual networks with the `mesh_peering_enabled` value set to `true`.
 
-variable "virtual_network_use_remote_gateways" {
-  type        = bool
-  description = <<DESCRIPTION
-Enables the use of remote gateways for the virtual network.
+- `mesh_peering_enabled`: Whether to enable mesh peering for this virtual network. Must be enabled on more than one virtual network for any peerings to be created. [optional]
+- `mesh_peering_allow_forwarded_traffic`: Whether to allow forwarded traffic for the mesh peering. [optional - default false]
 
-Applies to hub and spoke (vnet peerings).
-DESCRIPTION
-  default     = true
-}
+### Other peerings
 
-variable "virtual_network_vwan_associated_routetable_resource_id" {
-  type        = string
-  description = <<DESCRIPTION
-The resource ID of the virtual network route table to use for the virtual network.
+TODO
```

Comment:
I assume you are skipping this on the PR on purpose?

Reply:
Removed - thanks